Automattic/jetpack
# Fix post editor dirty state caused by publicize store

Opening the Jetpack sidebar on the post edit page results in dirty state of the post editor and thus preventing users from reloading or navigating away from the page.

<img width="508" alt="Screenshot 2023-11-13 at 12 39 17 PM" src="https://github.com/Automattic/jetpack/assets/18226415/3045eabb-e6ec-433d-9ff9-1c2e6c56f523">


## Proposed changes:

* Move the connections state and actions from `jetpack/publicize` to `jetpack-social-store` and update consumer components to use the new store.
* The source of truth for the connections will now be `jetpack-social-store` instead of post meta.
* Connections in post meta will only be synced to post meta when a user explicitly toggles a connection

**Note:**
After this PR, the store inside `projects/js-packages/publicize-components/src/store` technically contains the utility functions not the actual selectors or actions because they mostly talk to the `core/editor` store not the `jetpack/publicize` store. That means we can convert those selectors/actions to simple utility functions and get rid of `jetpack/publicize` store.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:

- Run `jetpack build plugins/jetpack`
- Ensure that you have some social connections added for the site
- Goto post edit page for an existing post and open Jetpack sidebar
- Wait for the `connection-test-results` API request to finish
- Try to reload the page
- Confirm that their is no confirmation popup about unsaved changes
- Toggle a connection
- Confirm that the UI reflects the toggled connection state
- Try to reload the page
- Confirm that this time you see the popup
- Confirm that the social preview details like name and profile picture work correctly
- Create a new post and publish it
- Confirm that it is posted to the social media connections
